org.adempiere.base/src/org/compiere/process/ServerProcessCtl.java

```diff
@@ -83,7 +85,19 @@ public static ServerProcessCtl process (ProcessInfo pi, Trx trx, boolean managed
 		{
 			try 
 			{ 
-				instance = new MPInstance(Env.getCtx(), pi.getAD_Process_ID(), pi.getTable_ID(), pi.getRecord_ID(), pi.getRecord_UU()); 
+				instance = new MPInstance(Env.getCtx(), pi.getAD_Process_ID(), pi.getTable_ID(), pi.getRecord_ID(), pi.getRecord_UU());
+				// Get PrintFormat
+				MPrintFormat format = (MPrintFormat)pi.getTransientObject();
+				if (format == null)
+					format = (MPrintFormat)pi.getSerializableObject();

```

Comment:
is this really needed ? any use case for this ?

Reply:
The use case for this is when you start document print for a jasper print format, without this code block the Print Format and Language are not set in the process audit.

If you mean why trying to get both transient, then if null also serializable object, it is used the same way in [ServerReportCtl](https://github.com/idempiere/idempiere/blame/master/org.adempiere.base/src/org/compiere/print/ServerReportCtl.java#L402), I thought it would be safer like this, but but for now I could not find any use case where SerializableObject is set here.

Your other change requests are addressed in the latest commit.

Reply:
Yes, the getserializableobject call is not needed here, please remove it.

Reply:
ok, done